src/server/protocol.ts

```diff
@@ -2700,6 +2701,54 @@ export interface InlayHintsResponse extends Response {
     body?: InlayHintItem[];
 }
 
+export interface MapCodeRequestArgs {
+    /**
+     * The files and changes to try and apply/map.
+     */
+    mappings: MapCodeRequestDocumentMapping[];
+
+    /**
+     * Edits to apply before performing the mapping.
+     */
+    updates?: FileCodeEdits[];

```

Comment:
They're coming from the editor. They're meant to be used for refinement, where VS has a temporary patch that isn't really applied to the textbuffer, but still wants the next step of the chat refinement to use what the current patch is meant to be.

This is basically only for VS, too, and isn't even used (yet), because it's hard, but VS Code doesn't need it at all because it just applies changes immediately.